kernel/src/process_load.rs

```diff
@@ -0,0 +1,791 @@
+
+use crate::capabilities::{ProcessApprovalCapability, ProcessManagementCapability};
+use crate::config;
+use crate::create_capability;
+use crate::debug;
+use crate::kernel::Kernel;
+use crate::platform::chip::Chip;
+use crate::process::Process;
+use crate::process_checking;
+use crate::process_checking::AppVerifier;
+use crate::process_policies::ProcessFaultPolicy;
+use crate::process_standard::ProcessStandard;
+use crate::static_init;
+use crate::utilities::cells::OptionalCell;
+use crate::ErrorCode;
+
+use tock_tbf::types::TbfFooterV2Credentials;
+use tock_tbf::types::TbfParseError;
+
+/// Errors that can occur when trying to load and create processes.
+pub enum ProcessLoadError {
+    /// No TBF header was found.
+    TbfHeaderNotFound,
+
+    /// The TBF header for the process could not be successfully parsed.
+    TbfHeaderParseFailure(tock_tbf::types::TbfParseError),
+
+    /// Not enough flash remaining to parse a process and its header.
+    NotEnoughFlash,
+
+    /// Not enough memory to meet the amount requested by a process. Modify the
+    /// process to request less memory, flash fewer processes, or increase the
+    /// size of the region your board reserves for process memory.
+    NotEnoughMemory,
+
+    /// A process was loaded with a length in flash that the MPU does not
+    /// support. The fix is probably to correct the process size, but this could
+    /// also be caused by a bad MPU implementation.
+    MpuInvalidFlashLength,
+
+    /// A process specified a fixed memory address that it needs its memory
+    /// range to start at, and the kernel did not or could not give the process
+    /// a memory region starting at that address.
+    MemoryAddressMismatch {
+        actual_address: u32,
+        expected_address: u32,
+    },
+
+    /// A process specified that its binary must start at a particular address,
+    /// and that is not the address the binary is actually placed at.
+    IncorrectFlashAddress {
+        actual_address: u32,
+        expected_address: u32,
+    },
+
+    /// A process requires a newer version of the kernel or did not specify
+    /// a required version. Processes can include the KernelVersion TBF header stating
+    /// their compatible kernel version (^major.minor).
+    ///
+    /// Boards may not require processes to include the KernelVersion TBF header, and
+    /// the kernel supports ignoring a missing KernelVersion TBF header. In that case,
+    /// this error will not be returned for a process missing a KernelVersion TBF
+    /// header.
+    ///
+    /// `version` is the `(major, minor)` kernel version the process indicates it
+    /// requires. If `version` is `None` then the process did not include the
+    /// KernelVersion TBF header.
+    IncompatibleKernelVersion { version: Option<(u16, u16)> },
+
+    /// The process did not contain a credentials which the process binary verifier
+    /// accepted and the verifier requires credentials.
+    CredentialsNoAccept,
```

Comment:
It might be they are supported by not accepted. E.g., there's a recognized/supported RSA-based signature but it's with a key the checker doesn't accept. I'll clarify.